gadget-container/entrypoint/entrypoint.go

```diff
@@ -0,0 +1,288 @@
+		return "", fmt.Errorf("opening file %s: %w", path, err)
+	}
+	defer file.Close()
+
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		line := scanner.Text()
+		parts := strings.SplitN(line, "=", 2)
+		if len(parts) != 2 {
+			continue
+		}
+
+		if parts[0] == "PRETTY_NAME" {
+			return strings.Trim(parts[1], "\""), nil
+		}
+	}
+
+	err = scanner.Err()
+	if err != nil {
+		return "", fmt.Errorf("reading file %s: %w", path, err)
+	}
+
+	return "", fmt.Errorf("%s does not contain PRETTY_NAME", path)
+}
+
+func getKernelRelease() (string, error) {
+	uts := &unix.Utsname{}
+	if err := unix.Uname(uts); err != nil {
+		return "", fmt.Errorf("calling uname: %w", err)
+	}
+
+	return unix.ByteSliceToString(uts.Release[:]), nil
+}
+
+func copyFile(destination, source string, filemode fs.FileMode) error {
+	content, err := os.ReadFile(source)
+	if err != nil {
+		return fmt.Errorf("reading %s: %w", source, err)
+	}
+
+	err = os.WriteFile(destination, content, filemode)
+	if err != nil {
+		return fmt.Errorf("writing %s: %w", destination, err)
+	}
+
+	return nil
+}
+
+func installCRIOHooks() error {
+	log.Info("Installing hooks scripts on host...")
+
+	path := filepath.Join(host.HostRoot, "opt/hooks/oci")
+	err := os.MkdirAll(path, 0o755)
+	if err != nil {
+		return fmt.Errorf("creating %s: %w", path, err)
+	}
+
+	for _, file := range []string{"ocihookgadget", "prestart.sh", "poststop.sh"} {
+		log.Infof("Installing %s", file)
+
+		path := filepath.Join("/opt/hooks/oci", file)
+		destinationPath := filepath.Join(host.HostRoot, path)
+		err := copyFile(destinationPath, path, 0o750)
+		if err != nil {
+			return fmt.Errorf("copying: %w", err)
+		}
+	}
+
+	for _, file := range []string{"etc/containers/oci/hooks.d", "usr/share/containers/oci/hooks.d/"} {
+		hookPath := filepath.Join(host.HostRoot, file)
+
+		log.Infof("Installing OCI hooks configuration in %s", hookPath)
+		os.MkdirAll(hookPath, 0o755)
+		if err != nil {
+			return fmt.Errorf("creating hook path %s: %w", path, err)
+		}
+
+		errCount := 0
+		for _, config := range []string{"/opt/hooks/crio/gadget-prestart.json", "/opt/hooks/crio/gadget-poststop.json"} {
+			hookFile := filepath.Join(hookPath, filepath.Base(config))
+			err := copyFile(hookFile, config, 0o640)
+			if err != nil {
+				errCount++
+			}
+		}
+
+		if errCount != 0 {
+			log.Warn("Couldn't install OCI hooks configuration")
+		} else {
+			log.Info("Hooks installation done")
+		}
+	}
+
+	return nil
+}
+
+func installNRIHooks() error {
+	log.Info("Installing NRI hooks")
+
+	destinationPath := filepath.Join(host.HostRoot, "opt/nri/bin")
+	err := os.MkdirAll(destinationPath, 0o755)
+	if err != nil {
+		return fmt.Errorf("creating %s: %w", destinationPath, err)
+	}
+
+	path := "/opt/hooks/nri/nrigadget"
+	destinationPath = filepath.Join(destinationPath, filepath.Base(path))
+	err = copyFile(destinationPath, path, 0o640)
+	if err != nil {
+		return fmt.Errorf("copying: %w", err)
+	}
+
+	configPath := "/opt/hooks/nri/conf.json"
```

Comment:
I think this needs to be [/host/etc/nri/conf.json](https://github.com/inspektor-gadget/inspektor-gadget/blob/main/gadget-container/entrypoint.sh#L96). I mean we should be appending our plugin only if host has existing config?